ObjectiveC plugin breaks windows build

-- libprotoc.vcproj doesn't contain objectivec generator files.
(fix for that one is here: https://github.com/jtattermusch/protobuf/tree/objectivec_partial_fix)

-- Use of posix headers:
Error	30	error C1083: Cannot open include file: 'arpa/inet.h': No such file or directory	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	31	1	libprotoc
Error	45	error C1083: Cannot open include file: 'netinet/in.h': No such file or directory	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	43	1	libprotoc

-- Use of int32_t
Error	47	error C2065: 'int32_t' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.h	168	1	libprotoc

-- possibly other problems

I got down to a point where I get these errors:
Error	9	error C2065: 'LOCK_EX' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	826	1	libprotoc
Error	2	error C2065: 'O_APPEND' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	810	1	libprotoc
Error	3	error C2065: 'O_CREAT' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	810	1	libprotoc
Error	1	error C2065: 'O_WRONLY' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	810	1	libprotoc
Error	6	error C2065: 'S_IRGRP' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	811	1	libprotoc
Error	7	error C2065: 'S_IROTH' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	811	1	libprotoc
Error	4	error C2065: 'S_IRUSR' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	811	1	libprotoc
Error	5	error C2065: 'S_IWUSR' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	811	1	libprotoc
Error	12	error C3861: 'close': identifier not found	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	840	1	libprotoc
Error	10	error C3861: 'flock': identifier not found	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	826	1	libprotoc
Error	8	error C3861: 'open': identifier not found	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	810	1	libprotoc
Error	11	error C3861: 'write': identifier not found	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	839	1	libprotoc
Warning	13	warning C4715: 'google::protobuf::compiler::objectivec::`anonymous namespace'::OrderGroupForFieldDescriptor' : not all control paths return a value	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_message.cc	122	1	libprotoc
Warning	14	warning C4715: 'google::protobuf::compiler::objectivec::`anonymous namespace'::PrimitiveArrayTypeName' : not all control paths return a value	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc	107	1	libprotoc
Warning	15	warning C4715: 'google::protobuf::compiler::objectivec::`anonymous namespace'::PrimitiveTypeName' : not all control paths return a value	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc	79	1	libprotoc


@thomasvl , do you think you will be able to take a look at them?  I can share the changes I have so far.

On Mon, May 18, 2015 at 12:45 PM, Jan Tattermusch <notifications@github.com>
wrote:

> I got down to a point where I get these errors:
> Error 9 error C2065: 'LOCK_EX' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 826 1 libprotoc
> Error 2 error C2065: 'O_APPEND' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 810 1 libprotoc
> Error 3 error C2065: 'O_CREAT' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 810 1 libprotoc
> Error 1 error C2065: 'O_WRONLY' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 810 1 libprotoc
> Error 6 error C2065: 'S_IRGRP' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 811 1 libprotoc
> Error 7 error C2065: 'S_IROTH' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 811 1 libprotoc
> Error 4 error C2065: 'S_IRUSR' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 811 1 libprotoc
> Error 5 error C2065: 'S_IWUSR' : undeclared identifier
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 811 1 libprotoc
> Error 12 error C3861: 'close': identifier not found
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 840 1 libprotoc
> Error 10 error C3861: 'flock': identifier not found
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 826 1 libprotoc
> Error 8 error C3861: 'open': identifier not found
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 810 1 libprotoc
> Error 11 error C3861: 'write': identifier not found
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc
> 839 1 libprotoc
> Warning 13 warning C4715: 'google::protobuf::compiler::objectivec::anonymous
> namespace'::OrderGroupForFieldDescriptor' : not all control paths return a
> value
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_message.cc
> 122 1 libprotoc
> Warning 14 warning C4715: 'google::protobuf::compiler::objectivec::anonymous
> namespace'::PrimitiveArrayTypeName' : not all control paths return a value
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc
> 107 1 libprotoc
> Warning 15 warning C4715:
> 'google::protobuf::compiler::objectivec::`anonymous
> namespace'::PrimitiveTypeName' : not all control paths return a value
> c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc
> 79 1 libprotoc
>
> @thomasvl <https://github.com/thomasvl> , do you think you will be able
> to take a look at them? I can share the changes I have so far.
>
>
>
Sure, just started looking at things now.

@thomasvl  I tried and it looks pretty good.

You're missing  #include <stdint.h> in  objectivec_helpers.h (to declare the type int32_t).

Also, I see this warning:
2>c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_message.cc(122): warning C4715: 'google::protobuf::compiler::objectivec::`anonymous namespace'::OrderGroupForFieldDescriptor' : not all control paths return a value


Besides that, I see an error when compiling tests project (but that seems to be unrelated)
Error	3	error C2385: ambiguous access of 'arena_'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\map_field_inl.h	215	1	tests

Error	44	error C2065: 'kAddUnderscore' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	923	1	libprotoc
Error	55	error C2065: 'kOpAllUpper' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	973	1	libprotoc
Error	57	error C2065: 'kOpAllUpper' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	986	1	libprotoc
Error	48	error C2065: 'kOpAsIs' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	932	1	libprotoc
Error	52	error C2065: 'kOpFirstLower' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	936	1	libprotoc
Error	50	error C2065: 'kOpFirstUpper' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	934	1	libprotoc
Error	11	error C2065: 'length' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	742	1	libprotoc
Error	12	error C2065: 'length' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	743	1	libprotoc
Error	60	error C2065: 'length' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	106	1	libprotoc
Error	61	error C2065: 'length' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	107	1	libprotoc
Error	62	error C2065: 'length' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	108	1	libprotoc
Error	41	error C2065: 'op' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	922	1	libprotoc
Error	43	error C2065: 'op' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	923	1	libprotoc
Error	45	error C2065: 'op' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	924	1	libprotoc
Error	46	error C2065: 'op' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	925	1	libprotoc
Error	42	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	922	1	libprotoc
Error	47	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	932	1	libprotoc
Error	49	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	934	1	libprotoc
Error	51	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	936	1	libprotoc
Error	53	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	947	1	libprotoc
Error	54	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	973	1	libprotoc
Error	56	error C2065: 'op_' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	986	1	libprotoc
Error	9	error C2065: 'uint32_t' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	742	1	libprotoc
Error	58	error C2065: 'uint32_t' : undeclared identifier	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	106	1	libprotoc
Error	13	error C2070: 'unknown-type': illegal sizeof operand	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	743	1	libprotoc
Error	63	error C2070: 'unknown-type': illegal sizeof operand	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	108	1	libprotoc
Error	20	error C2086: 'const int google::protobuf::compiler::objectivec::`anonymous-namespace'::DecodeDataBuilder::uint8_t' : redefinition	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	909	1	libprotoc
Error	25	error C2086: 'const int google::protobuf::compiler::objectivec::`anonymous-namespace'::DecodeDataBuilder::uint8_t' : redefinition	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	910	1	libprotoc
Error	30	error C2086: 'const int google::protobuf::compiler::objectivec::`anonymous-namespace'::DecodeDataBuilder::uint8_t' : redefinition	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	911	1	libprotoc
Error	35	error C2086: 'const int google::protobuf::compiler::objectivec::`anonymous-namespace'::DecodeDataBuilder::uint8_t' : redefinition	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	912	1	libprotoc
Error	16	error C2143: syntax error : missing ';' before '='	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	907	1	libprotoc
Error	21	error C2143: syntax error : missing ';' before '='	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	909	1	libprotoc
Error	26	error C2143: syntax error : missing ';' before '='	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	910	1	libprotoc
Error	31	error C2143: syntax error : missing ';' before '='	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	911	1	libprotoc
Error	36	error C2143: syntax error : missing ';' before '='	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	912	1	libprotoc
Error	15	error C2146: syntax error : missing ';' before identifier 'kAddUnderscore'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	907	1	libprotoc
Error	34	error C2146: syntax error : missing ';' before identifier 'kOpAllUpper'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	912	1	libprotoc
Error	19	error C2146: syntax error : missing ';' before identifier 'kOpAsIs'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	909	1	libprotoc
Error	29	error C2146: syntax error : missing ';' before identifier 'kOpFirstLower'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	911	1	libprotoc
Error	24	error C2146: syntax error : missing ';' before identifier 'kOpFirstUpper'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	910	1	libprotoc
Error	10	error C2146: syntax error : missing ';' before identifier 'length'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	742	1	libprotoc
Error	59	error C2146: syntax error : missing ';' before identifier 'length'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_field.cc	106	1	libprotoc
Error	40	error C2146: syntax error : missing ';' before identifier 'op'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	922	1	libprotoc
Error	38	error C2146: syntax error : missing ';' before identifier 'op_'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	954	1	libprotoc
Error	17	error C2238: unexpected token(s) preceding ';'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	907	1	libprotoc
Error	22	error C2238: unexpected token(s) preceding ';'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	909	1	libprotoc
Error	27	error C2238: unexpected token(s) preceding ';'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	910	1	libprotoc
Error	32	error C2238: unexpected token(s) preceding ';'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	911	1	libprotoc
Error	37	error C2238: unexpected token(s) preceding ';'	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	912	1	libprotoc
Error	14	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	907	1	libprotoc
Error	18	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	909	1	libprotoc
Error	23	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	910	1	libprotoc
Error	28	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	911	1	libprotoc
Error	33	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	912	1	libprotoc
Error	39	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	c:\users\jtattermusch\github\protobuf\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc	954	1	libprotoc

--- a/objectivec/DevTools/check_version_stamps.sh
+++ b/objectivec/DevTools/check_version_stamps.sh
@@ -25,7 +25,7 @@ readonly ConstantName=GOOGLE_PROTOBUF_OBJC_GEN_VERSION
 readonly PluginSrc="${ProtoRootDir}/src/google/protobuf/compiler/objectivec/objectivec_file.cc"
 readonly PluginVersion=$( \
     cat "${PluginSrc}" \
-        | sed -n -e "s:const int32_t ${ConstantName} = \([0-9]*\);:\1:p"
+        | sed -n -e "s:const int32 ${ConstantName} = \([0-9]*\);:\1:p"
 )
 
 if [[ -z "${PluginVersion}" ]] ; then

--- a/src/google/protobuf/compiler/objectivec/objectivec_extension.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_extension.cc
@@ -77,7 +77,6 @@ ExtensionGenerator::ExtensionGenerator(const string& root_class_name,
 ExtensionGenerator::~ExtensionGenerator() {}
 
 void ExtensionGenerator::GenerateMembersHeader(io::Printer* printer) {
-  WriteClassNameToClassList(root_class_and_method_name_);
   if (IsFiltered()) {
     printer->Print("// $filter_reason$\n\n", "filter_reason", filter_reason_);
     return;

--- a/src/google/protobuf/compiler/objectivec/objectivec_field.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_field.cc
@@ -39,10 +39,6 @@
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/stubs/strutil.h>
 
-#ifndef htonl
-#include <netinet/in.h>
-#endif
-
 namespace google {
 namespace protobuf {
 namespace compiler {
@@ -107,7 +103,7 @@ void SetCommonFieldVariables(const FieldDescriptor* descriptor,
   string field_options = descriptor->options().SerializeAsString();
   // Must convert to a standard byte order for packing length into
   // a cstring.
-  uint32_t length = htonl(field_options.length());
+  uint32 length = ghtonl(field_options.length());
   if (length > 0) {
     string bytes((const char*)&length, sizeof(length));
     bytes.append(field_options);

--- a/src/google/protobuf/compiler/objectivec/objectivec_file.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_file.cc
@@ -40,13 +40,14 @@
 #include <google/protobuf/stubs/strutil.h>
 #include <sstream>
 
+namespace google {
+namespace protobuf {
+
 // This is also found in GPBBootstrap.h, and needs to be kept in sync.  It
 // is the version check done to ensure generated code works with the current
 // runtime being used.
-const int32_t GOOGLE_PROTOBUF_OBJC_GEN_VERSION = 30000;
+const int32 GOOGLE_PROTOBUF_OBJC_GEN_VERSION = 30000;
 
-namespace google {
-namespace protobuf {
 namespace compiler {
 namespace objectivec {
 FileGenerator::FileGenerator(const FileDescriptor *file)

--- a/src/google/protobuf/compiler/objectivec/objectivec_generator.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_generator.cc
@@ -82,10 +82,6 @@ bool ObjectiveCGenerator::Generate(const FileDescriptor* file,
     file_generator.GenerateSource(&printer);
   }
 
-  if (!WriteClassList(error)) {
-    return false;
-  }
-
   return true;
 }
 

--- a/src/google/protobuf/compiler/objectivec/objectivec_helpers.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_helpers.cc
@@ -28,12 +28,6 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#include <arpa/inet.h>
-#include <errno.h>
-#include <limits.h>
-#include <stdlib.h>
-#include <unistd.h>
-
 #include <fstream>
 #include <iostream>
 #include <sstream>
@@ -46,14 +40,6 @@
 #include <google/protobuf/descriptor.pb.h>
 #include <google/protobuf/stubs/strutil.h>
 
-#ifndef htonl
-#include <netinet/in.h>
-#endif
-
-#ifndef O_EXLOCK
-#include <sys/file.h>
-#endif
-
 // NOTE: src/google/protobuf/compiler/plugin.cc makes use of cerr for some
 // error case, so it seem to be ok to use as a back door for errors.
 
@@ -65,7 +51,6 @@ namespace objectivec {
 namespace {
 
 hash_set<string> gClassWhitelist;
-stringstream gClassListStream;
 
 // islower()/isupper()/tolower()/toupper() change based on locale.
 
@@ -754,7 +739,7 @@ string DefaultValue(const FieldDescriptor* field) {
 
         // Must convert to a standard byte order for packing length into
         // a cstring.
-        uint32_t length = htonl(default_string.length());
+        uint32 length = ghtonl(default_string.length());
         string bytes((const char*)&length, sizeof(length));
         bytes.append(default_string);
         return "(NSData*)\"" + CEscape(bytes) + "\"";
@@ -807,60 +792,6 @@ string BuildCommentsString(const SourceLocation& location) {
         prefix + StringReplace(lines[i], "$", "&#36;", true) + suffix;
   }
   return final_comments;
-}
-
-bool WriteClassList(string* error) {
-  const char* file_name = getenv("GPB_CLASSLIST_PATH");
-  if (file_name != NULL) {
-#ifndef O_EXLOCK
-    int fd = open(file_name, O_WRONLY | O_APPEND | O_CREAT,
-                  (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH));
-#else
-    int fd = open(file_name, O_WRONLY | O_APPEND | O_EXLOCK | O_CREAT,
-                  (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH));
-#endif
-    if (fd == -1) {
-      if (error != NULL) {
-        stringstream err_stream;
-        err_stream << endl << file_name << ":0:0: error:"
-                   << "Unable to open (" << errno << ")";
-        *error = err_stream.str();
-      }
-      return false;
-    }
-#ifndef O_EXLOCK
-    if (flock(fd, LOCK_EX) < 0) {
-      if (error != NULL) {
-        stringstream err_stream;
-        err_stream << endl << file_name << ":0:0: error:"
-                   << "Unable to lock (" << errno << ")";
-        *error = err_stream.str();
-      }
-      return false;
-    }
-#endif
-    // Need a local to hold the list so the cstring stays valid for the
-    // write call.
-    const string& class_list_str = gClassListStream.str();
-    int write_out = write(fd, class_list_str.c_str(), class_list_str.length());
-    int close_out = close(fd);
-    if (write_out == -1 || close_out == -1) {
-      if (error != NULL) {
-        stringstream err_stream;
-        err_stream << endl << file_name << ":0:0: error:"
-                   << "Unable to write (" << errno << ")";
-        *error = err_stream.str();
-      }
-      return false;
-    }
-  }
-  return true;
-}
-
-void WriteClassNameToClassList(const string& name) {
-  if (gClassListStream.good()) {
-    gClassListStream << name << '\n';
-  }
 }
 
 bool InitializeClassWhitelist(string* error) {
@@ -917,7 +848,7 @@ bool FilterClass(const string& name) {
   return gClassWhitelist.size() > 0;
 }
 
-void TextFormatDecodeData::AddString(int32_t key,
+void TextFormatDecodeData::AddString(int32 key,
                                      const string& input_for_decode,
                                      const string& desired_output) {
   for (vector<DataEntry>::const_iterator i = entries_.begin();
@@ -973,14 +904,14 @@ class DecodeDataBuilder {
   }
 
  private:
-  static const uint8_t kAddUnderscore = 0b10000000;
+  static const uint8 kAddUnderscore = 0x80;
 
-  static const uint8_t kOpAsIs = 0b00000000;
-  static const uint8_t kOpFirstUpper = 0b01000000;
-  static const uint8_t kOpFirstLower = 0b00100000;
-  static const uint8_t kOpAllUpper = 0b01100000;
+  static const uint8 kOpAsIs        = 0x00;
+  static const uint8 kOpFirstUpper  = 0x40;
+  static const uint8 kOpFirstLower  = 0x20;
+  static const uint8 kOpAllUpper    = 0x60;
 
-  static const int kMaxSegmentLen = 0b00011111;
+  static const int kMaxSegmentLen     = 0x1f;
 
   void AddChar(const char desired) {
     ++segment_len_;
@@ -988,7 +919,7 @@ class DecodeDataBuilder {
   }
 
   void Push() {
-    uint8_t op = (op_ | segment_len_);
+    uint8 op = (op_ | segment_len_);
     if (need_underscore_) op |= kAddUnderscore;
     if (op != 0) {
       decode_data_ += (char)op;
@@ -1020,7 +951,7 @@ class DecodeDataBuilder {
 
   bool need_underscore_;
   bool is_all_upper_;
-  uint8_t op_;
+  uint8 op_;
   int segment_len_;
 
   string decode_data_;

--- a/src/google/protobuf/compiler/objectivec/objectivec_helpers.h
+++ b/src/google/protobuf/compiler/objectivec/objectivec_helpers.h
@@ -142,9 +142,6 @@ string BuildFlagsString(const vector<string>& strings);
 
 string BuildCommentsString(const SourceLocation& location);
 
-bool WriteClassList(string* error);
-void WriteClassNameToClassList(const string& name);
-
 bool InitializeClassWhitelist(string* error);
 bool FilterClass(const string& name);
 
@@ -154,7 +151,7 @@ class TextFormatDecodeData {
  public:
   TextFormatDecodeData() {}
 
-  void AddString(int32_t key, const string& input_for_decode,
+  void AddString(int32 key, const string& input_for_decode,
                  const string& desired_output);
   size_t num_entries() const { return entries_.size(); }
   string Data() const;
@@ -165,7 +162,7 @@ class TextFormatDecodeData {
  private:
   GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(TextFormatDecodeData);
 
-  typedef std::pair<int32_t, string> DataEntry;
+  typedef std::pair<int32, string> DataEntry;
   vector<DataEntry> entries_;
 };
 

--- a/src/google/protobuf/compiler/objectivec/objectivec_helpers_unittest.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_helpers_unittest.cc
@@ -150,7 +150,7 @@ TEST(ObjCHelper, TextFormatDecodeData_RawStrings) {
 
   EXPECT_EQ(4, decode_data.num_entries());
 
-  uint8_t expected_data[] = {
+  uint8 expected_data[] = {
       0x4,
       0x1, 0x0, 'z', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'I', 'J', 0x0,
       0x3, 0x0, 'a', 'b', 'c', 'd', 'e', 'z', 'g', 'h', 'I', 'J', 0x0,
@@ -175,7 +175,7 @@ TEST(ObjCHelper, TextFormatDecodeData_ByteCodes) {
 
   EXPECT_EQ(5, decode_data.num_entries());
 
-  uint8_t expected_data[] = {
+  uint8 expected_data[] = {
       0x5,
       // All as is (00 op)
       0x1,  0x0A, 0x0,

--- a/src/google/protobuf/compiler/objectivec/objectivec_message.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_message.cc
@@ -119,6 +119,9 @@ int OrderGroupForFieldDescriptor(const FieldDescriptor* descriptor) {
     case FieldDescriptor::TYPE_BOOL:
       return 1;
   }
+
+  GOOGLE_LOG(FATAL) << "Can't get here.";
+  return 0;
 }
 
 struct FieldOrderingByStorageSize {
@@ -302,8 +305,6 @@ void MessageGenerator::GenerateMessageHeader(io::Printer* printer) {
     }
     return;
   }
-
-  WriteClassNameToClassList(class_name_);
 
   if (IsFiltered()) {
     printer->Print("// $filter_reason$\n\n",

--- a/src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc
+++ b/src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc
@@ -76,6 +76,9 @@ const char* PrimitiveTypeName(const FieldDescriptor* descriptor) {
     case OBJECTIVECTYPE_MESSAGE:
       return NULL;
   }
+
+  GOOGLE_LOG(FATAL) << "Can't get here.";
+  return NULL;
 }
 
 const char* PrimitiveArrayTypeName(const FieldDescriptor* descriptor) {
@@ -104,6 +107,9 @@ const char* PrimitiveArrayTypeName(const FieldDescriptor* descriptor) {
     case OBJECTIVECTYPE_MESSAGE:
       return "";  // Want NSArray
   }
+
+  GOOGLE_LOG(FATAL) << "Can't get here.";
+  return NULL;
 }
 
 void SetPrimitiveVariables(const FieldDescriptor* descriptor,
@@ -154,7 +160,6 @@ void RepeatedPrimitiveFieldGenerator::FinishInitialization(void) {
     variables_["array_comment"] = "";
   }
 }
-
 
 }  // namespace objectivec
 }  // namespace compiler

--- a/vsprojects/libprotoc.vcproj
+++ b/vsprojects/libprotoc.vcproj
@@ -372,6 +372,54 @@
       <File RelativePath="..\src\google\protobuf\compiler\javanano\javanano_params.h"></File>
       <File RelativePath="..\src\google\protobuf\compiler\javanano\javanano_primitive_field.h"></File>
       <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_enum.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_enum_field.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_extension.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_field.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_file.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_generator.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_helpers.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_map_field.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_message.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_message_field.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_oneof.h"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.h"
+        >
+      </File>
+      <File
         RelativePath="..\src\google\protobuf\compiler\python\python_generator.h"
         >
       </File>
@@ -633,6 +681,54 @@
       </File>
       <File
         RelativePath="..\src\google\protobuf\compiler\javanano\javanano_primitive_field.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_enum.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_enum_field.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_extension.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_field.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_file.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_generator.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_map_field.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_message.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_message_field.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_oneof.cc"
+        >
+      </File>
+      <File
+        RelativePath="..\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc"
         >
       </File>
       <File

